File: 0278-first-bad-version/0278-first-bad-version.c
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

int firstBadVersion(int n) {
    unsigned int good = -1, bad = n, test;
    
    while (good+1 != bad){
        test = (good + bad)>>1;
        if (isBadVersion(test)) bad = test;
        else good = test;
    }
    return bad;
}

File: 0278-first-bad-version/0278-first-bad-version.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        good = -1
        bad = n
        while good+1 != bad:
            test = ((good + bad)>>1)
            if isBadVersion(test):
                bad = test
            else:
                good = test
        print(good)
        print(bad)
        return bad


        